### ISSUE TITLE
numpy.nonzero example ambiguity

### ISSUE THREAD

**Issue by jsh9** [numpy/numpy#8743]
The first example used in numpy.nonzero is ambiguous:

```
>>> x = np.eye(3)
>>> x
array([[ 1.,  0.,  0.],
       [ 0.,  1.,  0.],
       [ 0.,  0.,  1.]])
>>> np.nonzero(x)
(array([0, 1, 2]), array([0, 1, 2]))
```

Because the two tuple elements in the result are identical, so users may be confused about which tuple element corresponds to which dimension.

Instead, the following example can be used to eliminate any ambiguity:

```
>>> x = np.array([[1,0,0],[0,2,0],[1,1,0]])
>>> x
array([[1, 0, 0],
       [0, 2, 0],
       [1, 1, 0]])
>>> (array([0, 1, 2, 2], dtype=int64), array([0, 1, 0, 1], dtype=int64))
```

**Comment by WarrenWeckesser** [comment]
This is a good suggestion.  @jsh9, if you are familiar with git and github, you should consider submitting this change in a pull request.

**Comment by jsh9** [comment]
Haven't tried using "pull request" before. Let me try and learn it.

**Comment by eric-wieser** [comment]
Merged in #8850, subsequently fixed in #8871
